### DISCUSSION TITLE
Dynamically setup the topology using python

### DISCUSSION THREAD

**Discussion by alm5awi94** [omnetpp/omnetpp#1095] [Q&A]
I am having problem actually to setup my topology dynamically in omnetpp using python. I am using currently omnetpp version: 6.0.2. for testing i am using the routing example provided in the samples folder. Whenever i run the ini file. I am getting the following errors:

15:54:12 **** Incremental Build of configuration release for project routing ****
make MODE=release all 
node/L2Queue.cc
node/L2Queue.cc:91:26: error: no member named 'getStage' in 'omnetpp::cSimulation'
    if (getSimulation()->getStage() == cSimulation::STAGE_CLEANUP)
        ~~~~~~~~~~~~~~~  ^
node/L2Queue.cc:91:53: error: no member named 'STAGE_CLEANUP' in 'omnetpp::cSimulation'
    if (getSimulation()->getStage() == cSimulation::STAGE_CLEANUP)
                                       ~~~~~~~~~~~~~^
2 errors generated.
make: *** [Makefile:116: out/clang-release//node/L2Queue.o] Error 1
"make MODE=release all" terminated with exit code 2. Build might be incomplete.

**Comment by rhornig** [comment]
No, you are not. There are no such lines in that sample in omnetpp-6.0.2. You have probably modified that file and introduced the build error. And how this description relates to topology and python mentioned in the title?

**Reply by alm5awi94** [reply]
Thanks for answering. The lines i ve shown are the error messages on the console in OMNeT++ IDE. I am running currently omnetpp version 6.0.2. However, the example i am trying to run belongs to the master branch of omnetpp repository. which contains python scripts examples. For instance, FromCsv.py which is located in the following path:

([FromCsv.py (https://github.com/omnetpp/omnetpp/blob/master/samples/routing/networks/FromCsv.py)

This example Illustrates how to dynamically setup topology using python. which is something i am interested in. 
Is there any other way to setup topology in omnetpp dynamically through csv or json. without specifiing that in ned file?

**Reply by rhornig** [reply]
A lot happened between version 6.0.2 and the current master branch. Specifically the kernel was refactored to support python access. This is a major change hence it will be the base of the OMNeT++ 7.0 version. You cannot use the example from the master branch with the 6.0.2 version. The necessary APIs are just not there.
